crates/starknet-types-core/src/felt/mod.rs

```diff
@@ -47,23 +49,68 @@ use arbitrary::{self, Arbitrary, Unstructured};
 pub struct Felt(pub(crate) FieldElement<Stark252PrimeField>);
 
 /// A non-zero [Felt].
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone, Copy, Eq, PartialEq, PartialOrd, Ord)]
 pub struct NonZeroFelt(FieldElement<Stark252PrimeField>);
 
+impl NonZeroFelt {
+    /// Create a [NonZeroFelt] as a constant.
+    /// # Safety
+    /// If the value is zero will panic.
+    pub const fn from_raw(value: [u64; 4]) -> Self {
+        assert!(
+            value[0] != 0 || value[1] != 0 || value[2] != 0 || value[3] != 0,
+            "Felt is zero"
+        );
+        let value = Felt::from_raw(value);
+        Self(value.0)
+    }
+
+    /// [Felt] constant that's equal to 1.
+    pub const ONE: Self = Self::from_felt_unchecked(Felt(
+        FieldElement::<Stark252PrimeField>::from_hex_unchecked("1"),
+    ));
+
+    /// [Felt] constant that's equal to 2.
+    pub const TWO: Self = Self::from_felt_unchecked(Felt(
+        FieldElement::<Stark252PrimeField>::from_hex_unchecked("2"),
+    ));
+
+    /// [Felt] constant that's equal to 3.
+    pub const THREE: Self = Self::from_felt_unchecked(Felt(
+        FieldElement::<Stark252PrimeField>::from_hex_unchecked("3"),
+    ));
+
+    /// Maximum value of [Felt]. Equals to 2^251 + 17 * 2^192.
+    pub const MAX: Self =
+        Self::from_felt_unchecked(Felt(FieldElement::<Stark252PrimeField>::const_from_raw(
+            UnsignedInteger::from_limbs([544, 0, 0, 32]),
+        )));
+
+    /// 2 ** 251
+    pub const ELEMENT_UPPER_BOUND: Felt = Felt::from_raw([
```

Comment:
Shouldn't the type be `Self` aka `NonZeroFelt` rather than `Felt`

Reply:
And you can use Sef(Felt::ELEMENT_UPPER_BOUND) as impl

Reply:
removed